analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.cs

```diff
@@ -176,6 +175,51 @@ public void Execute_PersistSymbols_InsideBlock()
                 ((ILocalReferenceOperation)operation.Instance).Local;
         }
 
+        [TestMethod]
+        public void Execute_UnusedVariable_ClearedAfterBlock()
+        {
+            const string code = @"
+var first = boolParameter ? true : false;
+Tag(""BeforeLastUse""); 
+bool second = first;
+if(boolParameter)
+    boolParameter.ToString();
+Tag(""AfterLastUse"");
+";
+            ISymbol firstSymbol = null;
+            var postProcess = new PostProcessTestCheck(x =>
+            {
+                if (x.Operation.Instance.TrackedSymbol() is { } symbol)
+                {
+                    if (symbol.Name == "first")
+                    {
+                        firstSymbol = symbol;
+                    }
+                }
+                return x.State;
+            });
+            var validator = SETestContext.CreateCS(code, postProcess).Validator;
+            validator.TagStates("BeforeLastUse").Should().HaveCount(2).And.OnlyContain(x => x[firstSymbol] != null);
+            validator.TagStates("AfterLastUse").Should().HaveCount(1).And.OnlyContain(x => x[firstSymbol] == null);
+        }
+
+        [TestMethod]
+        public void Execute_CapturedVariable_NotCleared()
```

Comment:
This is not a capture.
```suggestion
        public void Execute_OuterMethodParameter_NotCleared()
```
We should add a test with capture too.